conductor-client/src/main/kotlin/com/openlattice/organizations/OrganizationMetadataEntitySetsService.kt

```diff
@@ -254,7 +257,7 @@ class OrganizationMetadataEntitySetsService(
         val organizationMetadataEntitySetIds = organizationService.getOrganizationMetadataEntitySetIds(organizationId)
         val datasetEntityKeyIds = getDatasetEntityKeyIds(
             organizationMetadataEntitySetIds,
-            tables.map { it.id }
+            tables.map { it.oid }
```

Comment:
this is going to break all existing data and duplicate everything (within the metadata entity sets) -- is that intentional?

Reply:
oids also will also be reused, so that means table data will be written to the same entities as old deleted data, which seems bad...

Reply:
It is not intentional to break all existing data and duplicate everything. The issue is that the table `oid` is currently in use in prod and I thought this change is least dangerous.

In this pull request, `addDataset` is calling `getDatasetEntityKeyId` with `table.oid`:

https://github.com/openlattice/openlattice/pull/76/files#diff-98470d2cb0c12678b30beee4b37ffcaec7dc44416b0a553c6cbceb89335f1256R305

However, `addDatasetsAndColumns` is calling `getDatasetEntityKeyIds` with `table.id`:

https://github.com/openlattice/openlattice/pull/76/files#diff-98470d2cb0c12678b30beee4b37ffcaec7dc44416b0a553c6cbceb89335f1256R244

Unfortunately, `addDataset` is being called by `BackgroundExternalDatabaseSyncingService` here:

https://github.com/openlattice/openlattice/blob/develop/indexer/src/main/kotlin/com/openlattice/BackgroundExternalDatabaseSyncingService.kt#L183

As far as I can tell, this means that all new tables that are being synced right now are reserving ekids using the table `oid` instead of the `id`, which I've confirmed by looking in the `sync_ids` table.

On the other hand, `addDatasetsAndColumns` is using the table `id`, which sounds like is the correct logic. However, `addDatasetsAndColumns` is only called by `importMetadata` in `OrganizationsController`, which is not really used anywhere as far as I know.

Alternatively, I can undo this change and update `addDataset` to use the table `id`, though I think that would require a migration.

Do you have another recommendation?

Reply:
I think we should definitely be using id for entityKeyId over oid. If there's a spot oid has been used in prod, we'll need to wipe that data and reintegrate because that means we have a ton of duplicates.

Reply:
Ok, I'll make the update in this pull request and will move the data conversation to slack.